Modules/System Tests/Email/src/EmailMessageUnitTest.Codeunit.al

```diff
@@ -522,6 +522,157 @@ codeunit 134689 "Email Message Unit Test"
+        Message.Create(Recipients, 'Test subject', 'Test body', true);
+        Message.SetBody('Changed test body');
+        // Verify
+        Assert.AreEqual('Changed test body', Message.GetBody(), 'A different body was expected');
+    end;
+
+    [Test]
+    [TransactionModel(TransactionModel::AutoRollback)]
+    procedure AppendToBodyTest()
+    var
+        Message: Codeunit "Email Message";
+        Recipients: List of [Text];
+        Result: List of [Text];
+        Index: Integer;
+    begin
+        // Initialize
+        PermissionsMock.Set('Email Edit');
+
+        // Exercise
+        Message.Create(Recipients, 'Test subject', 'Test body', true);
+        Message.AppendToBody(' extended');
+        // Verify
+        Assert.AreEqual('Test body extended', Message.GetBody(), 'A different body was expected');
+    end;
+
+    [Test]
+    [TransactionModel(TransactionModel::AutoRollback)]
+    procedure SetSubjectTest()
+    var
+        Message: Codeunit "Email Message";
+        Recipients: List of [Text];
+        Result: List of [Text];
+        Index: Integer;
+    begin
+        // Initialize
+        PermissionsMock.Set('Email Edit');
+
+        // Exercise
+        Message.Create(Recipients, 'Test subject', 'Test body', true);
+        Message.SetSubject('Changed test subject');
+        // Verify
+        Assert.AreEqual('Changed test subject', Message.GetSubject(), 'A different subject was expected');
+    end;
+
+    [Test]
+    [TransactionModel(TransactionModel::AutoRollback)]
+    procedure SetBodyHTMLFormattedTest()
+    var
+        Message: Codeunit "Email Message";
+        Recipients: List of [Text];
+        Result: List of [Text];
+        Index: Integer;
+    begin
+        // Initialize
+        PermissionsMock.Set('Email Edit');
+
+        // Exercise
+        Message.Create(Recipients, 'Test subject', 'Test body', true);
+        Message.SetBodyHTMLFormatted(false);
+        // Verify
+        Assert.IsFalse(Message.IsBodyHTMLFormatted(), 'Body was expected not to be HTML formatted');
+    end;
+
+    [Test]
+    [TransactionModel(TransactionModel::AutoRollback)]
+    procedure SetRecipientsToTest()
+    var
+        Message: Codeunit "Email Message";
+        Recipients: List of [Text];
+        Result: List of [Text];
+        Index: Integer;
+    begin
+        // Initialize
+        PermissionsMock.Set('Email Edit');
+
+        // Exercise
+        Message.Create(Recipients, 'Test subject', 'Test body', true);
+        Recipients.Add('recipient1@test.com');
+        Recipients.Add('recipient2@test.com');
+        Recipients.Add('recipient3@test.com');
+        Message.SetRecipients(Enum::"Email Recipient Type"::"To", Recipients);
+        Message.GetRecipients(Enum::"Email Recipient Type"::"To", Result);
+
+        // Verify
+        for Index := 1 to Result.Count() do
+            Assert.AreEqual(StrSubstNo(RecipientLbl, Index), Result.Get(Index), 'A different recipient was expected');
```

Comment:
I'd also like to see a test that sets the CC recipients (for instance) and checks that To and BCC are not changed.